### ISSUE TITLE
Add `string-trim-before-length-checks` rule

### ISSUE THREAD

**Issue by TheWandere** [marcalexiei/eslint-plugin-zod#253]
### 🧩 Description

This could also be part of the existing string trim rule.

It matters where trim is called and it should always be called before length checks like .min() and .max()
E.g.:
**z.string().trim().min(1)** with input of **a bunch of space chars** will fail because the min check is done after its already trimmed.
**z.string().min(1).trim()** with the **same input** will allow it because the space chars pass the min constrained and are trimmed afterwards.

I cant imagine a valid use case for somebody wanting this and i consider this an easy pitfall. 

### ⚙️ Options

N/A

### ✅ Valid Code Examples

```typescript
z.string().trim().min(1)
z.string().trim().length(1)
```

### ❌ Invalid Code Examples

```typescript
z.string().min(1).trim()
z.string().length(1).trim()
```

**Comment by marcalexiei** [comment]
Done via #269

**Comment by marcalexiei** [comment]
New [prefer-trim-before-string-length-checks](https://github.com/marcalexiei/eslint-plugin-zod/blob/main/docs/rules/prefer-trim-before-string-length-checks.md) rule available from [v3.12.0](https://github.com/marcalexiei/eslint-plugin-zod/releases/tag/v3.12.0)
